pyvista/plotting/composite_mapper.py

```diff
@@ -0,0 +1,1032 @@
+
+        # start with 1 as there is always a composite dataset and this is the
+        # root of the tree
+        cc = 1
+        for dataset in self._dataset:
+            if isinstance(dataset, MultiBlock):
+                cc += len(dataset) + 1  # include the block itself
+            else:
+                cc += 1
+        return cc
+
+    def __iter__(self):
+        """Return an iterator of all the block attributes."""
+        for ii in range(len(self)):
+            yield self[ii]
+
+
+class CompositePolyDataMapper(_vtk.vtkCompositePolyDataMapper2):
+    """Composite PolyData mapper.
+
+    Parameters
+    ----------
+    dataset : pyvista.MultiBlock
+        Multiblock dataset.
+
+    color_missing_with_nan : bool, optional
+        Color any missing values with the ``nan_color``. This is useful
+        when not all blocks of the composite dataset have the specified
+        ``scalars``.
+
+    interpolate_before_map : bool, optional
+        Enabling makes for a smoother scalars display.  Default is
+        ``True``.  When ``False``, OpenGL will interpolate the
+        mapped colors which can result is showing colors that are
+        not present in the color map.
+
+    """
+
+    def __init__(self, dataset, color_missing_with_nan=None, interpolate_before_map=None):
+        """Initialize this composite mapper."""
+        super().__init__()
+        self.SetInputDataObject(dataset)
+
+        # this must be added to set the color, opacity, and visibility of
+        # individual blocks
+        self._attr = CompositeAttributes(self, dataset)
+        self._dataset = dataset
+
+        if color_missing_with_nan is not None:
+            self.color_missing_with_nan = color_missing_with_nan
+        if interpolate_before_map is not None:
+            self.interpolate_before_map = interpolate_before_map
+
+    @property
+    def dataset(self) -> 'pv.MultiBlock':
+        """Return the composite dataset assigned to this mapper.
+
+        Examples
+        --------
+        >>> import pyvista as pv
+        >>> dataset = pv.MultiBlock([pv.Cube(), pv.Sphere(center=(0, 0, 1))])
+        >>> pl = pv.Plotter()
+        >>> actor, mapper = pl.add_composite(dataset)
+        >>> mapper.dataset   # doctest:+SKIP
+        MultiBlock (...)
+          N Blocks:     2
+          X Bounds:     -0.500, 0.500
+          Y Bounds:     -0.500, 0.500
+          Z Bounds:     -0.500, 1.500
+
+        """
+        return self._dataset
+
+    @property
+    def interpolate_before_map(self) -> bool:
+        """Return or set the interpolation of scalars before mapping.
+
+        Examples
+        --------
+        Disable interpolation before mapping.
+
+        >>> import pyvista as pv
+        >>> dataset = pv.MultiBlock([pv.Cube(), pv.Sphere(center=(0, 0, 1))])
+        >>> pl = pv.Plotter()
+        >>> actor, mapper = pl.add_composite(dataset)
+        >>> mapper.interpolate_before_map = False
+        >>> mapper.interpolate_before_map
```

Comment:
Similar almost-blank example figure. Especially with the link to the Interpolate before map example we can probably skip rendering this.

Reply:
Fixed in 8835c292f03a212cc953200b58f421e82e67c120.